File: app/src/main/java/com/android/sunshine/app/activities/MainActivity.java
package com.android.sunshine.app.activities;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import com.android.sunshine.app.App;
import com.android.sunshine.app.R;
import com.android.sunshine.app.adapter.ForecastCursorAdapter;
import com.android.sunshine.app.adapter.ItemClickCallback;
import com.android.sunshine.app.utils.ApplicationPreferences;
import com.android.sunshine.app.utils.Navigator;
import com.android.sunshine.app.utils.WeatherDetails;
import com.android.sunshine.app.weather.WeatherRepository;
import javax.inject.Inject;

public class MainActivity extends AppCompatActivity implements ItemClickCallback {

    @Inject
    Navigator navigator;

    @Inject
    ApplicationPreferences applicationPreferences;

    @Inject
    WeatherRepository weatherRepository;

    private boolean twoPane;

    @Override
    protected void onCreate(final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ((App) getApplication()).getComponent().inject(this);

        long dateFromUri = getIntent().getLongExtra("date", 0);
        if (dateFromUri != 0) {
            applicationPreferences.setInitialSelectedDate(dateFromUri);
            // TODO: mirar para k usamos InitialSelectedDate
        }

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayShowTitleEnabled(false);

        if (findViewById(R.id.weather_detail_container) != null) {
            twoPane = true;
            if (savedInstanceState == null) {
                navigator.twoPaneDetails(weatherRepository.getForecastForDetailWidget().get(0), this);
            }
        } else {
            twoPane = false;
        }
        applicationPreferences.useTodayLayout(!twoPane);
    }

    @Override
    public boolean onCreateOptionsMenu(final Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(final MenuItem item) {
        if (item.getItemId() == R.id.action_settings) {
            navigator.launchSettingsActivity(this);
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onItemSelected(final WeatherDetails weatherDetails, final ForecastCursorAdapter.ViewHolder viewHolder) {
        if (twoPane) {
            navigator.twoPaneDetails(weatherDetails, this);
        } else {
            navigator.transitionToDetails(this, weatherDetails, viewHolder.forecastIcon);
        }
    }
}

File: app/src/main/kotlin/com/android/sunshine/app/activities/MainActivity.kt
package com.android.sunshine.app.activities

import android.support.v7.app.ActionBarActivity
import com.android.sunshine.app.callbacks.ItemClickCallback
import android.os.Bundle
import com.android.sunshine.app.fragments.ForecastFragment
import android.view.Menu
import android.view.MenuItem
import com.android.sunshine.app.R
import com.android.sunshine.app.fragments.DetailFragment
import com.android.sunshine.app.sync.SyncAdapter
import android.content.Intent

public class MainActivity : ActionBarActivity(), ItemClickCallback {

    private var twoPane: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super<ActionBarActivity>.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        if (findViewById(R.id.fragment_detail_container) != null) {
            twoPane = true
            if (savedInstanceState == null) {
                getSupportFragmentManager().beginTransaction().replace(R.id.fragment_detail_container,
                        DetailFragment()).commit()
            }
        } else {
            twoPane = false
        }
        SyncAdapter.initializeSyncAdapter(this)
        val forecastFragment = getSupportFragmentManager().findFragmentById(
                R.id.fragment_forecast) as ForecastFragment
        forecastFragment.setUseTodayLayout(!twoPane)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        getMenuInflater().inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.getItemId() == R.id.action_settings) {
            startActivity(Intent(this, javaClass<SettingsActivity>()))
            return true
        }
        return super<ActionBarActivity>.onOptionsItemSelected(item)
    }

    override fun onItemSelected(date: String) {
        if (twoPane) {
            val args = Bundle()
            args.putString(DetailActivity().DATE_KEY, date)
            val detailFragment = DetailFragment()
            detailFragment.setArguments(args)
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_detail_container, detailFragment).commitAllowingStateLoss()
        } else {
            val intent = Intent(this, javaClass<DetailActivity>())
            intent.putExtra(DetailActivity().DATE_KEY, date)
            startActivity(intent)
        }
    }
}